packaging/suse/datadog-agent.init

```diff
@@ -0,0 +1,481 @@
+        datadog_supervisor_processes=$(echo "$supervisor_processes" |
+                                       grep -Ev $DD_OPT_PROC_REGEX |
+                                       grep $NAME |
+                                       grep -c RUNNING)
+
+        # Number of non-failed OPTIONAL supervisord programs (dogstatsd, jmxfetch and go-metro)
+        datadog_supervisor_opt_processes=$(echo "$supervisor_processes" |
+                                       grep -E $DD_OPT_PROC_REGEX |
+                                       grep $NAME |
+                                       grep -cv FATAL)
+
+        # Number of expected running supervisord programs (ignoring dogstatsd, jmxfetch and go-metro)
+        supervisor_config_programs=$(grep -Ev $DD_OPT_PROC_REGEX $SUPERVISOR_FILE |
+                                     grep -c '\[program:')
+
+        # Number of expected optional supervisord programs (dogstatsd, jmxfetch and go-metro)
+        supervisor_config_opt_programs=$(grep -E $DD_OPT_PROC_REGEX $SUPERVISOR_FILE |
+                                     grep -c '\[program:')
+
+        if [ "$datadog_supervisor_processes" -ne "$supervisor_config_programs" ]; then
+            ESSENTIAL=1
+        fi
+
+        if [ "$datadog_supervisor_opt_processes" -ne "$supervisor_config_opt_programs" ]  ; then
+            OPTIONAL=1
+        fi
+
+        if [ "$QUERY" = "essential" ] && [ "$ESSENTIAL" -eq 0 ]; then
+            FAIL=0
+        elif [ "$QUERY" = "optional" ] && [ "$OPTIONAL" -eq 0 ]; then
+            FAIL=0
+        elif [ "$QUERY" = "all" ] && [ "$ESSENTIAL" = "$OPTIONAL" ] && [ $ESSENTIAL -eq 0 ]; then
+            FAIL=0
+        fi
+
+        if [ "$FAIL" -gt 0 ]; then
+            echo "$supervisor_processes"
+            echo "$DESC (supervisor) is NOT running all child processes"
+            return 1
+        else
+            echo "$DESC (supervisor) is running all child processes"
+            return 0
+        fi
+
+
+    else
+        echo "$DESC (supervisor) is not running"
+        return 1
+    fi
+}
+
+# Action to take
+case "$1" in
+    start)
+        if [ ! -f $AGENTCONF ]; then
+            echo "$AGENTCONF not found. Exiting."
+            exit 3
+        fi
+
+        check_status essential > /dev/null
+        if [ $? -eq 0 ]; then
+            echo "$DESC is already running"
+            exit 0
+        else
+            $0 stop > /dev/null 2>&1
+        fi
+
+        su $AGENTUSER -c "$AGENTPATH configcheck" > /dev/null 2>&1
+        if [ $? -ne 0 ]; then
+            log_daemon_msg "Invalid check configuration. Please run sudo /etc/init.d/datadog-agent configtest for more details."
+            log_daemon_msg "Resuming starting process."
+        fi
+
+
+        log_daemon_msg "Starting $DESC (using supervisord)" "$NAME"
+        PATH=$SYSTEM_PATH /sbin/start-stop-daemon --start --quiet --oknodo --exec $SUPERVISORD_PATH -- -c $SUPERVISOR_FILE --pidfile $SUPERVISOR_PIDFILE
```

Comment:
Pretty crazy the PATH isn't enforced despite putting `/sbin` in it at the top of the script.